### PR TITLE
DOC-208: Revisions to JSDoc added by developers since July 2019

### PR DESCRIPTION
**Pull request by jennaingersoll**
https://github.com/highfidelity/hifi/pull/15974 - Sitting API
https://github.com/highfidelity/hifi/pull/16025 - Debug API for Picks/Pointers
https://github.com/highfidelity/hifi/pull/16144 - Avatar "look at" target
https://github.com/highfidelity/hifi/pull/16245 - Directional blending for point reaction
https://github.com/highfidelity/hifi/pull/16399 - Updated blend shape support with ARKit compatibility

See https://highfidelity.atlassian.net/browse/DOC-208

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Can someone from High Fidelity approve this PR?

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22533/HighFidelity-Beta-PR16492-ed9db81.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22533/HighFidelity-Beta-Interface-PR16492-ed9db81.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22533/HighFidelity-Beta-PR16492-ed9db81.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22533/HighFidelity-Quest-Beta-PR16492-ed9db81.apk)

File: assignment-client/src/avatars/ScriptableAvatar.h [hunk 1]
```diff
@@ -76,6 +76,22 @@
  *     size in the virtual world. <em>Read-only.</em>
  * @property {boolean} hasPriority - <code>true</code> if the avatar is in a "hero" zone, <code>false</code> if it isn't.
  *     <em>Read-only.</em>
+ * @property {boolean} hasScriptedBlendshapes=false - <code>true</code> if blendshapes are controlled by scripted actions, 
+ *     otherwise <code>false</code>. Set this to true before using the {@link MyAvatar.setBlendshape} method, and set back
```
Comment (ctrlaltdavid) [inline assignment-client/src/avatars/ScriptableAvatar.h]: <code>true</code>

File: interface/src/raypick/PickScriptingInterface.cpp [hunk 2]
```diff
@@ -245,9 +249,11 @@ std::shared_ptr<PickQuery> PickScriptingInterface::buildStylusPick(const QVarian
...
- * @property {PickType} pickType - The type of pick when getting these properties from {@link Picks.getPickProperties} or {@link Picks.getPickScriptParameters}. A parabola pick's type is {@link PickType.Parabola}.
- * @property {Vec3} baseScale - Returned from {@link Picks.getPickProperties} when the pick has a parent with varying scale (usually an avatar or an entity).
- *     Its value is the original scale of the parent at the moment the pick was created, and is used to rescale the pick, and/or the pointer which owns this pick, if any.
+ * @property {PickType} pickType - The type of pick when getting these properties from {@link Picks.getPickProperties} or 
+ *     {@link Picks.getPickScriptParameters}. A parabola pick's type is {@link PickType.Parabola}.
+ * @property {Vec3} baseScale - Returned from {@link Picks.getPickProperties} when the pick has a parent with varying scale 
+ *     (usually an avatar or an entity). Its value is the original scale of the parent at the moment the pick was created, and 
+ *     is used to rescale the pick, and/or the pointer which owns this pick, if any.
```
Comment (ctrlaltdavid) [inline interface/src/raypick/PickScriptingInterface.cpp]:
```suggestion
 *     is used to rescale the pick and the pointer which owns this pick, if any.
```

File: interface/src/raypick/PickScriptingInterface.h [hunk 3]
```diff
@@ -147,19 +147,20 @@ class PickScriptingInterface : public QObject, public Dependency {
...
-    * @param {number} id - The ID of the pick.
-    * @returns {Picks.RayPickProperties|Picks.ParabolaPickProperties|Picks.StylusPickProperties|Picks.CollisionPickProperties} User-provided properties, per the pick <code>type</code>.
-    */
+     * Gets the parameters that were passed in to {@link Picks.createPick} to create the pick, if the pick was created through 
+     * a script. Note that these properties do not reflect the current state of the pick.
+     * See {@link Picks.getPickProperties}.
+     * @function Picks.getPickScriptParameters
+     * @param {number} id - The ID of the pick.
+     * @returns {Picks.RayPickProperties|Picks.ParabolaPickProperties|Picks.StylusPickProperties|Picks.CollisionPickProperties} 
+     *     User-provided properties, per the pick <code>type</code>.
```
Comment (ctrlaltdavid) [inline interface/src/raypick/PickScriptingInterface.h]:
```suggestion
     *     Script-provided properties, per the pick <code>type</code>.
```

File: interface/src/raypick/PointerScriptingInterface.cpp [hunk 4]
```diff
@@ -89,10 +89,13 @@ QVariantMap PointerScriptingInterface::getPointerScriptParameters(unsigned int u
  * @property {Pointers.StylusPointerModel} [model] - Override some or all of the default stylus model properties.
  * @property {boolean} [hover=false] - <code>true</code> if the pointer generates {@link Entities} hover events, 
  *     <code>false</code> if it doesn't.
- * @property {PickType} pointerType - The type of pointer when getting these properties from {@link Pointers.getPointerProperties} or {@link Pointers.getPointerScriptParameters}. A stylus pointer's type is {@link PickType.Stylus}.
- * @property {number} [pickID] - Returned from {@link Pointers.getPointerProperties}. The ID of the pick created alongside this pointer.
+ * @property {PickType} pointerType - The type of the stylus pointer returned from {@link Pointers.getPointerProperties} 
+ *     or {@link Pointers.getPointerScriptParameters}. A stylus pointer's type is {@link PickType.Stylus}.
```
Comment (ctrlaltdavid) [inline interface/src/raypick/PointerScriptingInterface.cpp]:
```suggestion
 *     or {@link Pointers.getPointerScriptParameters}. A stylus pointer's type is {@link PickType(0)|PickType.Stylus}.
```

File: interface/src/raypick/PointerScriptingInterface.cpp [hunk 5]
```diff
@@ -208,8 +211,10 @@ std::shared_ptr<Pointer> PointerScriptingInterface::buildStylus(const QVariant&
  *     <code>false</code> if it doesn't.
  * @property {Pointers.Trigger[]} [triggers=[]] - A list of ways that a {@link Controller} action or function should trigger 
  *     events on the entity or overlay currently intersected.
- * @property {PickType} pointerType - The type of pointer when getting these properties from {@link Pointers.getPointerProperties} or {@link Pointers.getPointerScriptParameters}. A laser pointer's type is {@link PickType.Ray}.
- * @property {number} [pickID] - Returned from {@link Pointers.getPointerProperties}. The ID of the pick created alongside this pointer.
+ * @property {PickType} pointerType - The type of pointer returned from {@link Pointers.getPointerProperties} or 
+ *     {@link Pointers.getPointerScriptParameters}. A laser pointer's type is {@link PickType.Ray}.
```
Comment (ctrlaltdavid) [inline interface/src/raypick/PointerScriptingInterface.cpp]:
```suggestion
 *     {@link Pointers.getPointerScriptParameters}. A laser pointer's type is {@link PickType(0)|PickType.Ray}.
```

File: interface/src/raypick/PointerScriptingInterface.cpp [hunk 6]
```diff
@@ -401,8 +406,10 @@ std::shared_ptr<Pointer> PointerScriptingInterface::buildLaserPointer(const QVar
  *     <code>false</code> if it doesn't.
  * @property {Pointers.Trigger[]} [triggers=[]] - A list of ways that a {@link Controller} action or function should trigger
  *     events on the entity or overlay currently intersected.
- * @property {PickType} pointerType - The type of pointer when getting these properties from {@link Pointers.getPointerProperties} or {@link Pointers.getPointerScriptParameters}. A parabola pointer's type is {@link PickType.Parabola}.
- * @property {number} [pickID] - Returned from {@link Pointers.getPointerProperties}. The ID of the pick created alongside this pointer.
+ * @property {PickType} pointerType - The type of pointer returned from {@link Pointers.getPointerProperties} or 
+ *     {@link Pointers.getPointerScriptParameters}. A parabola pointer's type is {@link PickType.Parabola}.
```
Comment (ctrlaltdavid) [inline interface/src/raypick/PointerScriptingInterface.cpp]:
```suggestion
 *     {@link Pointers.getPointerScriptParameters}. A parabola pointer's type is {@link PickType(0)|PickType.Parabola}.
```

File: interface/src/raypick/PointerScriptingInterface.h [hunk 7]
```diff
@@ -161,13 +161,14 @@ class PointerScriptingInterface : public QObject, public Dependency {
...
-    * See {@link Pointers.getPointerProperties}.
+    * Gets the parameters that were passed in to {@link Pointers.createPointer} to create the pointer when the pointer was 
+    * created through a script. 
+    * <p><strong>Note:</strong> These properties do not reflect the current state of the pointer. To get the current state 
+    * of the pointer, see {@link Pointers.getPointerProperties}.
     * @function Pointers.getPointerScriptParameters
     * @param {number} id - The ID of the pointer.
-    * @returns {Pointers.RayPointerProperties|Picks.ParabolaPointerProperties|Picks.StylusPointerProperties} User-provided properties, per the pointer <code>type</code>.
+    * @returns {Pointers.RayPointerProperties|Picks.ParabolaPointerProperties|Picks.StylusPointerProperties} 
+    *     User-provided properties, per the pointer <code>type</code>.
```
Comment (ctrlaltdavid) [inline interface/src/raypick/PointerScriptingInterface.h]:
```suggestion
    *     Script-provided properties, per the pointer <code>type</code>.
```

File: libraries/avatars/src/AvatarData.h [hunk 8]
```diff
@@ -532,19 +532,22 @@ class AvatarData : public QObject, public SpatiallyNestable {
...
+     *     facial blendshapes, <code>false</code> if automatic blinking is disabled. Set this property to false if you wish to
+     *     fully control the blink facial blendshapes via the {@link MyAvatar.setBlendshape} method.
+     * @property {boolean} hasProceduralEyeFaceMovement=true - <code>true</code> if the facial blendshapes for an avatar's eyes 
+     *     adjust automatically as the eyes move, <code>false</code> if this automatic movement is disabled. Set this property 
+     *     to true to prevent the iris from being obscured by the upper or lower lids. Set this property to <code>false</code> 
+     *     if you wish to fully control the eye blendshapes via the {@link MyAvatar.setBlendshape} method.
+     * @property {boolean} hasAudioEnabledFaceMovement=true - <code>true</code> if the avatar's mouth blendshapes animate 
+     *     automatically based on detected microphone input, <code>false</code> if this automatic movement is disabled. Set 
+     *     this property to false if you wish to fully control the mouth facial blendshapes via the 
+     *     {@link MyAvatar.setBlendshape} method.
```
Comment (ctrlaltdavid) [inline libraries/avatars/src/AvatarData.h]: `<code>true</code>`, `<code>false</code>`, and `<strong>Note:</strong>` per previously.

File: libraries/avatars/src/AvatarData.h [hunk 9]
```diff
@@ -1130,9 +1133,9 @@ class AvatarData : public QObject, public SpatiallyNestable {
 
 
     /**jsdoc
-     * Sets the value of a blendshape to animate your avatar's face. To enable other users to see the resulting animation of 
-     * your avatar's face, set {@link Avatar.hasScriptedBlendshapes} to true while using this API and back to false when your
-     * animation is complete.
+     * Sets the value of a blendshape to animate your avatar's face. In order for other users to see the resulting animations 
+     * on your avatar's face, set {@link Avatar.hasScriptedBlendshapes} to true. When you are done using this API, set
```
Comment (ctrlaltdavid) [inline libraries/avatars/src/AvatarData.h]:
```suggestion
     * on your avatar's face, set {@link Avatar.hasScriptedBlendshapes} to <code>true</code>. When you are done using this API, set 
```

File: libraries/avatars/src/AvatarData.h [hunk 10]
```diff
@@ -1130,9 +1133,9 @@ class AvatarData : public QObject, public SpatiallyNestable {
 
 
     /**jsdoc
-     * Sets the value of a blendshape to animate your avatar's face. To enable other users to see the resulting animation of 
-     * your avatar's face, set {@link Avatar.hasScriptedBlendshapes} to true while using this API and back to false when your
-     * animation is complete.
+     * Sets the value of a blendshape to animate your avatar's face. In order for other users to see the resulting animations 
+     * on your avatar's face, set {@link Avatar.hasScriptedBlendshapes} to true. When you are done using this API, set 
+     * {@link Avatar.hasScriptedBlendshapes} back to false when the animation is complete.
```
Comment (ctrlaltdavid) [inline libraries/avatars/src/AvatarData.h]:
```suggestion
     * {@link Avatar.hasScriptedBlendshapes} back to <code>false</code> when the animation is complete. 
```

File: assignment-client/src/avatars/ScriptableAvatar.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (jennaingersoll) [reply assignment-client/src/avatars/ScriptableAvatar.h]: Done.

File: assignment-client/src/avatars/ScriptableAvatar.h [hunk 11]
```diff
@@ -76,6 +76,22 @@
  *     size in the virtual world. <em>Read-only.</em>
  * @property {boolean} hasPriority - <code>true</code> if the avatar is in a "hero" zone, <code>false</code> if it isn't.
  *     <em>Read-only.</em>
+ * @property {boolean} hasScriptedBlendshapes=false - <code>true</code> if blendshapes are controlled by scripted actions, 
+ *     otherwise <code>false</code>. Set this to true before using the {@link MyAvatar.setBlendshape} method, and set back 
+ *     to false after you no longer want scripted control over the blendshapes.
```
[the remark this answers is not in the document]
Comment (jennaingersoll) [reply assignment-client/src/avatars/ScriptableAvatar.h]: Done.

File: assignment-client/src/avatars/ScriptableAvatar.h [hunk 12]
```diff
@@ -76,6 +76,22 @@
...
+ *     <p>NOTE: This property will automatically be set to true if the Controller system has valid facial blendshape
```
[the remark this answers is not in the document]
Comment (jennaingersoll) [reply assignment-client/src/avatars/ScriptableAvatar.h]: Done.

File: assignment-client/src/avatars/ScriptableAvatar.h [hunk 13]
```diff
@@ -76,6 +76,22 @@
  *     size in the virtual world. <em>Read-only.</em>
  * @property {boolean} hasPriority - <code>true</code> if the avatar is in a "hero" zone, <code>false</code> if it isn't.
  *     <em>Read-only.</em>
+ * @property {boolean} hasScriptedBlendshapes=false - <code>true</code> if blendshapes are controlled by scripted actions, 
+ *     otherwise <code>false</code>. Set this to true before using the {@link MyAvatar.setBlendshape} method, and set back 
+ *     to false after you no longer want scripted control over the blendshapes.
+ *     <p>NOTE: This property will automatically be set to true if the Controller system has valid facial blendshape 
+ *     actions.</p>
+ * @property {boolean} hasProceduralBlinkFaceMovement=true - <code>true</code> if avatars blink automatically by animating 
+ *     facial blendshapes, <code>false</code> if automatic blinking is disabled. Set this property to false if you wish to
```
[the remark this answers is not in the document]
Comment (jennaingersoll) [reply assignment-client/src/avatars/ScriptableAvatar.h]: Done.

File: assignment-client/src/avatars/ScriptableAvatar.h [hunk 14]
```diff
@@ -76,6 +76,22 @@
...
+ *     fully control the blink facial blendshapes via the {@link MyAvatar.setBlendshape} method.
+ * @property {boolean} hasProceduralEyeFaceMovement=true - <code>true</code> if the facial blendshapes for an avatar's eyes 
+ *     adjust automatically as the eyes move, <code>false</code> if this automatic movement is disabled. Set this property 
+ *     to true to prevent the iris from being obscured by the upper or lower lids. Set this property to <code>false</code>
```
[the remark this answers is not in the document]
Comment (jennaingersoll) [reply assignment-client/src/avatars/ScriptableAvatar.h]: Done.

File: assignment-client/src/avatars/ScriptableAvatar.h [hunk 15]
```diff
@@ -76,6 +76,22 @@
...
+ * @property {boolean} hasProceduralBlinkFaceMovement=true - <code>true</code> if avatars blink automatically by animating 
+ *     facial blendshapes, <code>false</code> if automatic blinking is disabled. Set this property to false if you wish to
+ *     fully control the blink facial blendshapes via the {@link MyAvatar.setBlendshape} method.
+ * @property {boolean} hasProceduralEyeFaceMovement=true - <code>true</code> if the facial blendshapes for an avatar's eyes 
+ *     adjust automatically as the eyes move, <code>false</code> if this automatic movement is disabled. Set this property 
+ *     to true to prevent the iris from being obscured by the upper or lower lids. Set this property to <code>false</code> 
+ *     if you wish to fully control the eye blendshapes via the {@link MyAvatar.setBlendshape} method.
+ * @property {boolean} hasAudioEnabledFaceMovement=true - <code>true</code> if the avatar's mouth blendshapes animate 
+ *     automatically based on detected microphone input, <code>false</code> if this automatic movement is disabled. Set 
+ *     this property to false if you wish to fully control the mouth facial blendshapes via the
```
[the remark this answers is not in the document]
Comment (jennaingersoll) [reply assignment-client/src/avatars/ScriptableAvatar.h]: Done.

File: interface/src/avatar/MyAvatar.h [hunk 16]
```diff
@@ -148,6 +148,22 @@ class MyAvatar : public Avatar {
...
+     *     facial blendshapes, <code>false</code> if automatic blinking is disabled. Set this property to false if you wish to
+     *     fully control the blink facial blendshapes via the {@link MyAvatar.setBlendshape} method.
+     * @property {boolean} hasProceduralEyeFaceMovement=true - <code>true</code> if the facial blendshapes for an avatar's eyes 
+     *     adjust automatically as the eyes move, <code>false</code> if this automatic movement is disabled. Set this property 
+     *     to true to prevent the iris from being obscured by the upper or lower lids. Set this property to <code>false</code> 
+     *     if you wish to fully control the eye blendshapes via the {@link MyAvatar.setBlendshape} method.
+     * @property {boolean} hasAudioEnabledFaceMovement=true - <code>true</code> if the avatar's mouth blendshapes animate 
+     *     automatically based on detected microphone input, <code>false</code> if this automatic movement is disabled. Set 
+     *     this property to false if you wish to fully control the mouth facial blendshapes via the 
+     *     {@link MyAvatar.setBlendshape} method.
```
[the remark this answers is not in the document]
Comment (jennaingersoll) [reply interface/src/avatar/MyAvatar.h]: Done.

File: interface/src/avatar/MyAvatar.h [hunk 17]
```diff
@@ -1746,7 +1762,7 @@ class MyAvatar : public Avatar {
     void prepareAvatarEntityDataForReload();
 
     /**jsdoc
-    * Turn the avatar's head until it faces the target point within the 90/-90 degree range.
+    * Turns the avatar's head until it faces the target point within a 90/-90 degree range.
```
[the remark this answers is not in the document]
Comment (jennaingersoll) [reply interface/src/avatar/MyAvatar.h]: Done.

File: interface/src/avatar/MyAvatar.h [hunk 18]
```diff
@@ -1755,47 +1771,46 @@ class MyAvatar : public Avatar {
...
     * @function MyAvatar.releaseHeadLookAtControl
     */
     Q_INVOKABLE void releaseHeadLookAtControl();
 
     /**jsdoc
-    * Force the avatar's eyes to look to the specified location.
-    * Once this method is called, API calls will have full control of the eyes for a limited time.
-    * If this method is not called for two seconds, the engine will regain control of the eyes.
+    * Forces the avatar's eyes to look at a specified location. Once this method is called, API calls 
+    * will have full control of the eyes for a limited time. If this method is not called for two seconds,
```
[the remark this answers is not in the document]
Comment (jennaingersoll) [reply interface/src/avatar/MyAvatar.h]: Done.

File: interface/src/avatar/MyAvatar.h [hunk 19]
```diff
@@ -1755,47 +1771,46 @@ class MyAvatar : public Avatar {
...
     */
     Q_INVOKABLE void releaseHeadLookAtControl();
 
     /**jsdoc
-    * Force the avatar's eyes to look to the specified location.
-    * Once this method is called, API calls will have full control of the eyes for a limited time.
-    * If this method is not called for two seconds, the engine will regain control of the eyes.
+    * Forces the avatar's eyes to look at a specified location. Once this method is called, API calls 
+    * will have full control of the eyes for a limited time. If this method is not called for two seconds, 
+    * the engine will regain control of the eyes.
```
[the remark this answers is not in the document]
Comment (jennaingersoll) [reply interface/src/avatar/MyAvatar.h]: Done.

File: interface/src/avatar/MyAvatar.h [hunk 20]
```diff
@@ -1755,47 +1771,46 @@ class MyAvatar : public Avatar {
...
     /**jsdoc
-    * Aims the pointing directional blending towards the provided target point.
-    * The "point" reaction should be triggered before using this method. 
-    * <code>MyAvatar.beginReaction("point")</code>
-    * Returns <code>true</code> if the target point lays in front of the avatar.
+    * Aims the pointing directional blending towards the provided target point. The "point" reaction should be triggered 
+    * before using this method with the code <code>MyAvatar.beginReaction("point")</code>.
     * @function MyAvatar.setPointAt
     * @param {Vec3} pointAtTarget - The target point in world coordinates.
+    * @returns {boolean} <code>true</code> if the target point lays in front of the avatar; <code>false</code> otherwise.
```
[the remark this answers is not in the document]
Comment (jennaingersoll) [reply interface/src/avatar/MyAvatar.h]: Done.

File: interface/src/raypick/PickScriptingInterface.cpp (hunk 2 above)
Comment (jennaingersoll) [reply interface/src/raypick/PickScriptingInterface.cpp]: Done.

File: interface/src/raypick/PickScriptingInterface.h (hunk 3 above)
Comment (jennaingersoll) [reply interface/src/raypick/PickScriptingInterface.h]: Done.

File: interface/src/raypick/PointerScriptingInterface.cpp (hunk 4 above)
Comment (jennaingersoll) [reply interface/src/raypick/PointerScriptingInterface.cpp]: Done.

File: interface/src/raypick/PointerScriptingInterface.cpp (hunk 5 above)
Comment (jennaingersoll) [reply interface/src/raypick/PointerScriptingInterface.cpp]: Done.

File: interface/src/raypick/PointerScriptingInterface.cpp (hunk 6 above)
Comment (jennaingersoll) [reply interface/src/raypick/PointerScriptingInterface.cpp]: Done.

File: interface/src/raypick/PointerScriptingInterface.h (hunk 7 above)
Comment (jennaingersoll) [reply interface/src/raypick/PointerScriptingInterface.h]: Done.

File: libraries/avatars/src/AvatarData.h (hunk 8 above)
Comment (jennaingersoll) [reply libraries/avatars/src/AvatarData.h]: Done.

File: libraries/avatars/src/AvatarData.h (hunk 9 above)
Comment (jennaingersoll) [reply libraries/avatars/src/AvatarData.h]: Done.

File: libraries/avatars/src/AvatarData.h (hunk 10 above)
Comment (jennaingersoll) [reply libraries/avatars/src/AvatarData.h]: Done.

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22618/HighFidelity-Beta-PR16492-d2118f0.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22618/HighFidelity-Quest-Beta-PR16492-d2118f0.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22618/HighFidelity-Beta-PR16492-d2118f0.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22618/HighFidelity-Beta-Interface-PR16492-d2118f0.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/22618/HighFidelity-Beta-PR16492-d2118f0.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/22618/HighFidelity-Beta-Interface-PR16492-d2118f0.exe

**Pull request merged**